app/javascript/components/molecules/ExpandableTable.js

```diff
@@ -0,0 +1,79 @@
+import React, { Fragment } from 'react'
+
+class ExpandableTableRow extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {row: this.props.row}
+  }
+
+  toggleExpandedRow = () => {
+    this.setState((prevState) => {
+      return {
+        row: {
+          ...prevState.row,
+          expanded: !prevState.row.expanded
```

Comment:
we work on this together, but I just realized a problem. This is confusing and might create a problem:

`this.state = {row: this.props.row}` 
and then updating the state in the row like 
`row {...., expanded: !prevState.row.expanded }`

The bad practice here is to the set `this.props.row` as a default state since changes in the parent will not be propagated down. I would leave `this.props.row` as property and create a `this.state.expanded` attribute and we track the expanded from there. Not sure if you need `expanded` outside the component, but if we do we need to create an even like on expanded. In way, the row gets cleaner since it does not need to know about the expanded state.

Usually when setting state from props is for setting default state values. The best practice is to name the property as `defaulBlaBla` so the person using that component know that subsequent changes in that property will not get reflected in the view.

Reply:
ah i see, thanks for catching that! yes, let’s make the change the way you described. i’ll add the change to this pr